flink-cyber/metron-parser-chain/parser-chains-config-service/src/main/java/com/cloudera/parserchains/queryservice/service/KafkaService.java

```diff
@@ -1,84 +1,77 @@
+    public Pair<ResponseType, ResponseBody> sendWithReply(RequestType requestType, String clusterId, RequestBody body) {
+        final ClouderaReplyingKafkaTemplate<String, RequestBody, ResponseBody> kafkaTemplate = kafkaTemplatePool.get(clusterId);
+        if (kafkaTemplate == null) {
+            log.error("Cluster not found with cluster id: '{}'", clusterId);
+            throw new KafkaClusterNotFound("Cluster not found! with cluster id '" + clusterId + "'");
+        }
+        return send(requestType, body, clusterId, kafkaTemplate);
     }
 
-    final String jsonBody = JSONUtils.INSTANCE.toJSON(body, false);
-
-    ProducerRecord<String, String> record = new ProducerRecord<>(kafkaTemplate.getRequestTopic(),
-        messageType.name(), jsonBody);
-    RequestReplyFuture<String, String, String> replyFuture = kafkaTemplate.sendAndReceive(record);
-    try {
-      ConsumerRecord<String, String> consumerRecord = replyFuture.get(15, TimeUnit.SECONDS);
-      if (consumerRecord == null) {
-        throw new KafkaException("Got no reply");
-      }
-      final String stringReply = consumerRecord.value();
-
-      return deserializeReply(stringReply, responseReader);
-    } catch (InterruptedException | ExecutionException | TimeoutException e) {
-      // Handle exception
-      throw new KafkaReplyTimeoutException("Timeout while waiting for reply");
-    } catch (IOException e) {
-      throw new RuntimeException("Wasn't able to deserialize the response!", e);
+    public List<Pair<ResponseType, ResponseBody>> sendWithReply(RequestType requestType, RequestBody body) {
+        return kafkaTemplatePool.entrySet().stream()
+                .map(entry -> send(requestType, body, entry.getKey(), entry.getValue()))
+                .collect(Collectors.toList());
     }
-  }
-
-  private static <T> ResponseEntity<T> deserializeReply(String reply, ObjectReader responseReader) throws IOException {
-    final JsonNode replyBody = MAPPER.readTree(reply);
-
-    final T body = responseReader == null
-        ? null
-        : responseReader.readValue(replyBody.get(BODY_PARAM));
-    final HttpHeaders headers = HEADERS_READER.readValue(replyBody.get(HEADERS_PARAM));
-    final HttpStatus status = HttpStatus.valueOf(replyBody.get(STATUS_PARAM).asText());
-
-    return new ResponseEntity<>(body, headers, status);
-  }
 
+    private Pair<ResponseType, ResponseBody> send(RequestType requestType, RequestBody body, String clusterId, ClouderaReplyingKafkaTemplate<String, RequestBody, ResponseBody> kafkaTemplate) {
+        ProducerRecord<String, RequestBody> producerRecord = new ProducerRecord<>(kafkaTemplate.getRequestTopic(), requestType.name(), body);
+        RequestReplyFuture<String, RequestBody, ResponseBody> replyFuture = kafkaTemplate.sendAndReceive(producerRecord, Duration.ofMinutes(1));
+        try {
+            ConsumerRecord<String, ResponseBody> consumerRecord = replyFuture.get(45, TimeUnit.SECONDS);
```

Comment:
We should have this timeout in properties so that it can be changed from outside